### PR TITLE
Porting visualizations to new gallery

### PR DESCRIPTION
**Pull request by suhaani-agarwal**
porting the following visualizations from https://rcdata.nau.edu/genomic-ml/animint-gallery/ to https://animint.github.io/gallery/

43 - [Montreal Bikes Jan 2016](https://suhaani-agarwal.github.io/montreal-bikes-Jan2016/) , old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2016-01-24-Montreal-bikes-Jan-2016/index.html)
31 - [Demo of Constrained PDPA](https://suhaani-agarwal.github.io/Demo_Constrained_PDPA/), old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2017-02-20-constrained-PDPA-normal-real/index.html)
30 - [Cost/slack minimization over all feature thresholds, for building max margin interval tree models](https://suhaani-agarwal.github.io/figure-penaltyLearning/) , old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2017-03-22-Cost-slack-minimization-over-all-feature-thresholds,-for-building-max-margin-interval-tree-models/index.html)
50 - [Segmentation model selection with BIC](https://suhaani-agarwal.github.io/intreg-BIC/) , old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2014-12-16-Segmentation-model-selection-with-BIC/index.html)
5 - [Gradient descent for neural network and linear model](https://suhaani-agarwal.github.io/Ch18-neural-networks-sim/) , old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2023-01-30-neural-networks-sim/index.html) 
6 - [Gradient descent for regression](https://suhaani-agarwal.github.io/figure-gradient-descent-regression/) , old viz : [link](https://rcdata.nau.edu/genomic-ml/animint-gallery/2022-02-02-gradient-descent-regression/index.html)

I have also added the [video](https://vimeo.com/1087397034) for [ChIP sequencing data viz](https://suhaani-agarwal.github.io/chip-seq-unaligned/) ported in the previous PR. Please check if its ok.

here is the preview in my forked version of the gallery : https://suhaani-agarwal.github.io/gallery/

### CODE REVIEW

**Comment by tdhock** [comment]
Thanks!
mostly everything looks good but please adjust text content or text size or plot size so that text is readable (not overlapping other text).
image

**Comment by tdhock** [comment]
here there is a text ovelap between y axis ticks.
image
please fix by adjusting `theme(panel.margin=grid::unit(1, "lines"))` etc

**Comment by suhaani-agarwal** [comment]
Done, I have made these changes

**Comment by tdhock** [comment]
for https://suhaani-agarwal.github.io/intreg-BIC/ the animation (time option) is ok but can you please delete the duration option (smooth transition between model sizes does not make sense here / creates artificial visual noise such as changepoints that fly around randomly)

**Comment by suhaani-agarwal** [comment]
> for https://suhaani-agarwal.github.io/intreg-BIC/ the animation (time option) is ok but can you please delete the duration option (smooth transition between model sizes does not make sense here / creates artificial visual noise such as changepoints that fly around randomly)

Got it, Done!

**Comment by suhaani-agarwal** [comment]
> for https://suhaani-agarwal.github.io/intreg-BIC/ the animation (time option) is ok but can you please delete the duration option (smooth transition between model sizes does not make sense here / creates artificial visual noise such as changepoints that fly around randomly)

Got it, done!

**Comment by tdhock** [comment]
please fix text overlaps in facet titles, 
image
 by adjusting `theme(panel.margin=grid::unit(1, "lines"))` etc

**Comment by suhaani-agarwal** [comment]
image

It was not overlapping for me....But I have added the margin in between to avoid the text overlap. Is it okay now?  https://suhaani-agarwal.github.io/intreg-BIC/

**Comment by tdhock** [comment]
please try firefox in which text is bigger than chrome for example.
image
maybe change the text content, or text size, or `theme_animint(height=bigger_number_of_pixels)`

**Comment by suhaani-agarwal** [comment]
I have increased the plot height now, please check. this is what it shows for me on firefox:

image

**Comment by tdhock** [comment]
much better
image
thanks!
